crates/storage/libmdbx-rs/src/txn_manager.rs

```diff
@@ -155,27 +133,23 @@ mod read_transactions {
         }
 
         /// Adds a new transaction to the list of active read transactions.
-        pub(crate) fn add_active_read_transaction(&self, ptr: *mut ffi::MDBX_txn) {
+        pub(crate) fn add_active_read_transaction(
+            &self,
+            ptr: *mut ffi::MDBX_txn,
+            tx: TransactionPtr,
```

Comment:
how come both? I thought `tx` replaces need for `ptr`

Reply:
I think it's a bit unsafe. how about calling it with `TransactionPtr` and adding a getter on that type to read the ptr type? that way we can be certain that we are actually removing the right ptr from active.

Reply:
we don't want to add a getter of an internal pointer to `TransactionPtr` specifically because it's not safe to use it without locking the Mutex first, and the locking is done in `TransactionPtr::txn_execute`.

> that way we can be certain that we are actually removing the right ptr from active.

I'm not convinced it's a problem, if you remove a different ptr from active, it's an API misuse.

Reply:
ok good enough

Reply:
sure but at least that way, you are sure it's the same pointer, I don't see how it's less safe to pass it as a separate param

Reply:
but nvm, think not so many people touch this code